src/main/java/net/projectzombie/region_rotation/modules/StateBuffer.java

```diff
@@ -107,11 +107,16 @@ protected BaseState readBaseState(String regionName, World world)
             // Adding all the altStates
             if (altStateIDs != null)
             {
-                for (String altStateID : altStateIDs)
+                for (String altStateID : altStateIDs) {
                     baseState.addAltState(BaseState.toRegion(altStateID),
                             BaseState.toWorldUID(altStateID));
+                    baseState.changeStateBroadcast(BaseState.toRegion(altStateID),
+                                                   BaseState.toBroadcastMsg(altStateID));
+                }
             }
-            baseState.rotateState(currentState, false); // Don't erase built things.
+            baseState.rotateState(BaseState.toRegion(currentState), false); // Don't erase built things.
```

Comment:
I do not like BaseState.toRegion - we should not have to operate on strings to pass RegionStates to critical functions.